key_management.md

```diff
@@ -8,77 +8,96 @@ css_id: key_management
-Since the root role keys are expected to be revoked and replaced relatively rarely, its metadata file MAY expire after a relatively long time, such as one year.
 
 ## What to do in case of key compromise
-An OEM and its suppliers SHOULD be prepared to handle a key compromise. If the recommended number and type of keys are used, this should be a rare event. Nevertheless, when it happens OEMs and suppliers could use the following recovery procedures.
 
+An OEM and its suppliers SHOULD be prepared to handle a key compromise. If the recommended number and type of keys are used, this should be a rare event. Nevertheless, when it happens OEMs and suppliers could use the following recovery procedures.
 
 ### Director repository
 
 Since the Director repository MUST keep at least some software signing keys online, a compromise of this repository can lead to some security threats, such as mix-and-match attacks. Thus, the OEM SHOULD take great care to protect this repository, and reduce its attack surface as much as possible. For example, this MAY be done, in part, by using a firewall. However, if the repository has been compromised, then the following procedure SHOULD be performed in order to recover ECUs from the compromise.
 
 First, the OEM SHOULD use the Root role to revoke and replace the keys to the Timestamp, Snapshot, and Targets roles, because only the Root role can replace these keys. Following the type and placement of keys prescribed for the Director repository, we assume that attackers have compromised the online keys to the Timestamp, Snapshot, and Targets roles, but not the offline keys to the Root role.
 
-Second, the OEM SHOULD manually recall all vehicles in order to replace these keys. This MAY be done by requiring vehicle owners to visit the nearest dealership. Even though it could replace these keys on full verification ECU by using over-the-air broadcasts, a manual recall SHOULD be done because:
-1. the OEM SHOULD perform a safety inspection of the vehicles, in case of security attacks,and
-2. partial verification secondaries are not designed to handle key revocation and replacement over-the-air. In order to update keys for partial verification secondaries, the OEM should overwrite their copies of the root metadata file, perhaps using new images.
+Second, the OEM SHOULD manually recall all vehicles in order to replace these keys. This MAY be done by requiring vehicle owners to visit the nearest dealership. Even though it could replace these keys on a full verification ECU by using over-the-air broadcasts, a manual recall SHOULD be done because:
```

Comment:
I think this might be overly cautious. If there aren't partial-verification secondaries in place, I don't think it's productive to describe a recall as a SHOULD.

Reply:
I've reworded this section a bit more. I didn't come up with the original `SHOULD` but it seems reasonable to me in general, so I've tried to make it more clear when it might not be as necessary.